cmd/promtool/backfill_test.go

```diff
@@ -0,0 +1,263 @@
+// Copyright 2020 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"io/ioutil"
+	"math"
+	"os"
+	"path"
+	"path/filepath"
+	"testing"
+
+	"github.com/prometheus/prometheus/pkg/labels"
+	"github.com/prometheus/prometheus/storage"
+	"github.com/prometheus/prometheus/tsdb"
+	"github.com/prometheus/prometheus/tsdb/tsdbutil"
+	"github.com/prometheus/prometheus/util/testutil"
+)
+
+type backfillSample struct {
+	t int64
+	v float64
+}
+
+func (s backfillSample) T() int64   { return s.t }
+func (s backfillSample) V() float64 { return s.v }
+
+func createTemporaryOpenmetricsFile(t *testing.T, omFile string, text string) error {
+	f, err := os.Create(omFile)
+	testutil.Ok(t, err)
+	_, errW := f.WriteString(text)
+	testutil.Ok(t, errW)
+	testutil.Ok(t, f.Close())
+	return nil
+}
+
+func queryblock(t testing.TB, q storage.Querier, matchers ...*labels.Matcher) map[string][]tsdbutil.Sample {
```

Comment:
I have taken this from https://github.com/prometheus/prometheus/blob/master/tsdb/db_test.go#L78
Let me know if I should reuse these functions by putting them in a common file. I will also have to change the name query in db_test(which is used in many places in tsdb) to something else as promtool has it's own query function.